### ISSUE TITLE
Crash when applying 3DLUT

### ISSUE THREAD

**Issue by shun-iwasawa** [opentoonz/opentoonz#1862]
For now using 3DLUT color correction feature causes crash when displaying Style Editor / Color Model / Flipbook. 
It is probably due to the recent fix for the blank viewer. The 3DLUT feature used global OpenGL context but the previous fix disabled it.

**Comment by jpturcotte** [comment]
@shun-iwasawa Is a move to modern OpenGL (3.2+) still being considered? Thank you!

**Comment by Orphanlast** [comment]
Can I get this in Layman's terms so that I can report this story in an Opentoonz News video. OpenGL optimization interests me and it would interes my 850+ subscribers.

**Comment by shun-iwasawa** [comment]
@jpturcotte
Yes, but for now I interrupt it for tackling with other urgent modifications. I can post my intermediate work if someone will be interested in continuing it.
I'm a little worried about that moving to the core profile will not directly speed-up the software, according to [this slide](https://www.slideshare.net/Mark_Kilgard/gtc-2010-opengl/97).

@Orphanlast 
They are all related to [OpenGL Context](https://www.khronos.org/opengl/wiki/OpenGL_Context).
It is still very cumbersome for me so I'm afraid I cannot explain it properly..
In short, the story is as follows:
- OpenGL resources, such as textures, framebuffer objects (sort of rendered image data), and vertex buffer objects (vertex data) is associated with context.
- Every Viewers have separate contexts. By default, only if the Viewers are added as children to the same top-level widget (i.e. docked in the main window), their contexts will share with each other.
- When using the shader fxs it is rendered in the offscreen buffer which has another context.
- If you move between contexts or use resource across contexts you need some treatment. The shader fx did not work in the past due to the context problem.

- In v1.2.0 I introduced "[Global Share Context](http://doc.qt.io/qt-5/qopenglcontext.html#globalShareContext)" in order to avoid being bothered by such context handling.
- However, after releasing v1.2.0 the blank viewer issue was reported from users. It turned out to be due to the Global Share Context. Some computer seemed not to support it. So I stop using it.

- The 3DLUT feature is originally developed under Global Share Context environment. It needed to be fixed or OT crashes.
